data/completion/etelpmoc.sh

```diff
@@ -0,0 +1,180 @@
+    \compgen "$@"
+}
+alias compgen=xcompgen
+shopt -s expand_aliases
+
+# compopt replaces the original compopt with one that just sets/unsets entries
+# in _compopts
+compopt() (
+    local i
+
+    for ((i=0; i<$#; i++)); do
+        case "${!i}" in
+            -o)
+                ((i++))
+                _compopts[${!i}]=1
+                ;;
+            +o)
+                ((i++))
+                unset _compopts[${!i}]
+                ;;
+        esac
+    done
+)
+
+_compfunc="_minimal"
+_compact=""
+# this is a lot more complicated than it should be, but it's how you
+# get the result of 'complete -p "$1"' into an array, splitting it as
+# the shell would.
+readarray -t _comp < <(xargs -n1 < <(complete -p "$1") )
+if [[ "${_comp[*]}" ]]; then
+    while getopts :abcdefgjksuvA:C:W:o:F: opt "${_comp[@]:1}"; do
+        case "$opt" in
+            a)
+                _compact="alias"
+                ;;
+            b)
+                _compact="builtin"
+                ;;
+            c)
+                _compact="command"
+                ;;
+            d)
+                _compact="directory"
+                ;;
+            e)
+                _compact="export"
+                ;;
+            f)
+                _compact="file"
+                ;;
+            g)
+                _compact="group"
+                ;;
+            j)
+                _compact="job"
+                ;;
+            k)
+                _compact="keyword"
+                ;;
+            s)
+                _compact="service"
+                ;;
+            u)
+                _compact="user"
+                ;;
+            v)
+                _compact="variable"
+                ;;
+            A)
+                _compact="$OPTARG"
+                ;;
+            o)
+                _compopts["$OPTARG"]=1
+                ;;
+            C|F)
+                _compfunc="$OPTARG"
+                ;;
+            W)
+                readarray -t COMPREPLY < <( \compgen -W "$OPTARG" -- "${COMP_WORDS[$COMP_CWORD]}" )
+                _compfunc=""
+                ;;
+            *)
+                # P, G, S, and X are not supported yet
+                _die "ERROR: unknown option -$OPTARG"
+                ;;
+        esac
+    done
+fi
+
+_bounce=""
+case "$_compact" in
+    # these are for completing things that'll be interpreted by the
+    # "outside" bash, so send them back to be completed there.
+    "alias"|"export"|"job"|"variable")
+        _bounce="$_compact"
+        ;;
+esac
+
+if [ ! "$_bounce" ]; then
+    if [ "$_compact" ]; then
+        readarray -t COMPREPLY < <( \compgen -A "$_compact" -- "${COMP_WORDS[$COMP_CWORD]}" )
+    elif [ "$_compfunc" ]; then
+        # execute completion function (or the command if -C)
+        $_compfunc
+    fi
+fi
+
+# print completions to stdout
+echo "${!_compopts[@]}"
+echo "$_bounce"
+echo ""
+printf "%s\n" "${COMPREPLY[@]}"
```

Comment:
ignoring the one-single-printf-won't-work-for-this but looking at your request for `%q`, I'm not sure I'm following your reasoning here. The output of `etelpmoc.sh` isn't read as shell input, that is, it isn't just sourced from the 'outside'; it's used to set environment variables which the programmable completion tells you to set (or to call `compopt`). As such, if etelpmoc.sh quotes things, complete.sh needs to unquote them again, otherwise the user won't be offered the right completions.